src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -42,7 +47,14 @@ const TopicFragment = ({
 
   const isReady = useIsReady();
 
-  const topics = discovery.initialTopics;
+  const route = useRoute();
+
+  const topics = React.useMemo(() => {
+    return discovery.initialTopics.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [discovery.initialTopics]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `topics` variable is now memoized using `React.useMemo`. This is a good practice as it prevents unnecessary re-computations of the `topics` array when the component re-renders, improving performance.
```diff
- const topics = discovery.initialTopics;
+ const route = useRoute();
+ const topics = React.useMemo(() => {
+   return discovery.initialTopics.map((item) => ({
+     ...item,
+     following: item.following !== undefined ? item.following : item.user_id_follower !== null
+   }));
+ }, [discovery.initialTopics]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->